### DISCUSSION TITLE
Upgrade Options to Maintain DB

### DISCUSSION THREAD

**Discussion by fullhack2** [blakeblackshear/frigate#2115] [Ask A Question]
Hi all,

Any recommendations on how to best transition/save existing DB's when upgrading to a new image?  I have just been rebuilding my container with the new image each time, but was thinking it would be nice to keep the DB if possible (my clips and snapshots are stored on a network share, but the DB is running locally).

Or is it recommended to start with a fresh DB when moving to each new version?

Thanks!

**Comment by jasonpstokes** [comment]
Take a look at the docs on Storage, [here](https://docs.frigate.video/installation/#common-docker-compose-storage-configurations) (point 3) to get an idea of the database path and how to change it. But by default the database is stored in `/media/frigate/frigate.db` - the same location as your clips/recordings - so you should be able to do whatever you want with the container and still keep your database and recordings.

So if that's not your experience, can you post your docker container and Frigate config?

**Reply by fullhack2** [reply]
I believe I have solved my issue.

I had initially moved the entire /media/frigate structure to a CIFS share, but kept running into DB locking issues.

I then split that out so that just the clips and recordings were being directed to the CIFS share (as well as my config.yml so I could easily edit and restart):

> volumes:
>       - /etc/localtime:/etc/localtime:ro
>       - /media/d_drive/cameras/config.yml:/config/config.yml:ro
>       - /media/d_drive/cameras/clips:/media/frigate/clips
>       - /media/d_drive/cameras/recordings:/media/frigate/recordings
>      

This was working fine (keeping the DB on the container itself, but every update I would lose the old DB (unless I did some kind of manual backup and restore to the DB).

I just reconfigured my Docker compose file to redirect the /media/frigate directory to a perpetual volume on the Docker host and it does seem to be working and should allow me to keep the DB files after an upgrade, as well as my clips/recordings on the CIFS share, which is what I want:

> volumes:
      - /etc/localtime:/etc/localtime:ro
      - /media/d_drive/cameras/config.yml:/config/config.yml:ro
      - /media/d_drive/cameras/clips:/media/frigate/clips
      - /media/d_drive/cameras/recordings:/media/frigate/recordings
      - /var/lib/docker/volumes/frigate_data/_data:/media/frigate

**Reply by blakeblackshear** [reply]
Seem most recent docs about storage: https://docs.frigate.video/installation#storage
